File: anagram/main_test.py
# Execute: python3 -m unittest -v main_test.py

from main import Solution

import unittest

class TestSolution(unittest.TestCase):

    def test_1(self):
        sol = Solution()
        s = "anagram"
        t = "nagaram"
        expected_output = True
        self.assertEqual(sol.isAnagram(s, t), expected_output)

    def test_2(self):
        sol = Solution()
        s = "rat"
        t = "car"
        expected_output = False
        self.assertEqual(sol.isAnagram(s, t), expected_output)

if __name__ == '__main__':
    unittest.main()

File: arrays_and_hashing_problems/anagram/main_test.go
// go test -v
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type TestCase struct {
	testName       string
	s1             string
	s2             string
	expectedOutput bool
}

var testCases = []TestCase{
	{
		testName:       "Test1",
		s1:             "anagram",
		s2:             "nagaram",
		expectedOutput: true,
	},
	{
		testName:       "Test2",
		s1:             "rat",
		s2:             "car",
		expectedOutput: false,
	},
}

func TestSolutionWithCounterMap(t *testing.T) {
	assert := assert.New(t)

	for _, testCase := range testCases {
		t.Run(testCase.testName, func(t *testing.T) {
			s1 := testCase.s1
			s2 := testCase.s2
			expectedOutput := testCase.expectedOutput
			output := solutionWithCounterMap(s1, s2)
			assert.Equal(expectedOutput, output)
		})
	}
}

func TestSolutionWithCounterArray(t *testing.T) {
	assert := assert.New(t)

	for _, testCase := range testCases {
		t.Run(testCase.testName, func(t *testing.T) {
			s1 := testCase.s1
			s2 := testCase.s2
			expectedOutput := testCase.expectedOutput
			output := solutionWithCounterArray(s1, s2)
			assert.Equal(expectedOutput, output)
		})
	}
}
